api/src/main/java/org/apache/iceberg/types/PruneColumns.java

```diff
@@ -24,13 +24,27 @@
 import org.apache.iceberg.Schema;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.types.Types.ListType;
+import org.apache.iceberg.types.Types.MapType;
+import org.apache.iceberg.types.Types.StructType;
 
 class PruneColumns extends TypeUtil.SchemaVisitor<Type> {
   private final Set<Integer> selected;
+  private final boolean selectFullTypes;
 
-  PruneColumns(Set<Integer> selected) {
+  /**
+   * Visits a schema and returns only those element's whose id's have been passed as selected
+   * @param selected ids of elements to return
+   * @param selectFullTypes whether explicitly selected fields should automatically include all subfields
+   */
+  PruneColumns(Set<Integer> selected, boolean selectFullTypes) {
     Preconditions.checkNotNull(selected, "Selected field ids cannot be null");
     this.selected = selected;
+    this.selectFullTypes = selectFullTypes;
+  }
+
+  PruneColumns(Set<Integer> selected) {
```

Comment:
Since this is package-private, can we remove it and always explicitly pass `selectFullTypes`?